### PR TITLE
IPC Broadcast upload extension screenshare

### PR DESCRIPTION
**Pull request by davidliu**

### CODE REVIEW

File: common/darwin/Classes/FlutterRTCMediaStream.m [hunk 1]
```diff
@@ -490,10 +492,22 @@ -(void)getDisplayMedia:(NSDictionary *)constraints
     RTCMediaStream *mediaStream = [self.peerConnectionFactory mediaStreamWithStreamId:mediaStreamId];
 
     RTCVideoSource *videoSource = [self.peerConnectionFactory videoSource];
-    FlutterRPScreenRecorder *screenCapturer = [[FlutterRPScreenRecorder alloc] initWithDelegate:videoSource];
```
Comment (davidliu) [inline common/darwin/Classes/FlutterRTCMediaStream.m:493]: Should we have an option for using the in app capturer vs. broadcast upload extension capturer?

**Review by davidzhao — APPROVED** [review]
lgtm!

**Comment by cloudwebrtc** [comment]
cool, @davidliu  I can't wait to test this feature, do I need to add some code to example/ios/Runner?

**Comment by davidliu** [comment]
@cloudwebrtc the instructions are the same as found here:

https://jitsi.github.io/handbook/docs/dev-guide/dev-guide-ios-sdk/#screen-sharing-integration

**Review by hiroshihorie — APPROVED** [review]

**Review by cloudwebrtc — APPROVED** [review]

**Comment by davidzhao** [comment]
We will need to update our docs with examples and steps as next steps.

**Comment by cloudwebrtc** [comment]
Using Broadcast extension would be more complicated than just calling getDisplayMedia, so, I will try to make a simple example on flutter,  and next update the docs.

**Comment by davidzhao** [comment]
> We will need to update our docs with examples and steps as next steps.

oops, sorry, I should've been more clear. would be great to update the docs and example apps for livekit :) Forgot which repo I was in.

**Comment by davidliu** [comment]
Yeah, I have the broadcast extension implemented on the livekit-flutter side. I'm going to add an extra constraint option to access the broadcast extension capture so that this doesn't break existing users.

**Comment by JiaoLiu** [comment]
ios 14.x close broadcast screenshare will recieve servral alerts.

**Comment by JiaoLiu** [comment]
这个功能好像只支持手机之间分享屏幕（iOS->Android, iOS -> iOS），iPhone 投屏到Mac上失败。

**Comment by JiaoLiu** [comment]
> 

@cloudwebrtc

**Comment by cloudwebrtc** [comment]
应该可以的，可以尝试一下使用vp8， 我怀疑是h264 的默认 profile-devle-id 设置问题。

**Comment by JiaoLiu** [comment]
> Hmm, mac should be able to support h264 though right?
> 
> @JiaoLiu can you open a separate issue so we can discuss there?

#1151

**Pull request merged**